src/kakarot/accounts/eoa/aa/library.cairo

```diff
@@ -0,0 +1,197 @@
+            calldata=calldata + [call_array].data_offset,
+            );
+
+        is_valid_eth_tx(eth_address, _call.calldata_len, _call.calldata);
+
+        return validate(
+            eth_address=eth_address,
+            call_array_len=call_array_len - 1,
+            call_array=call_array + CallArray.SIZE,
+            calldata_len=calldata_len,
+            calldata=calldata,
+        );
+    }
+
+    // @notice decodes evm tx and validates it
+    // @dev 1. decodes the tx list
+    // @dev 2. recodes the list without the signature
+    // @dev 3. hashes the tx
+    // @dev 4. verifies the signature
+    // @dev TODO https://github.com/Flydexo/kakarot-eth-aa/issues/6
+    // @param eth_address The ethereum address owning the account
+    // @param calldata_len The lenght of the calldata
+    // @param calldata The calldata
+    // @return is_valid 1 if the transaction is valid
+    func is_valid_eth_tx{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        bitwise_ptr: BitwiseBuiltin*,
+        range_check_ptr,
+    }(eth_address: felt, calldata_len: felt, calldata: felt*) -> (is_valid: felt) {
+        alloc_locals;
+        let tx_type = [calldata];
+        // remove the tx type
+        let rlp_data = calldata + 1;
+        let (local items: RLP.Item*) = alloc();
+        // decode the rlp array
+        RLP.decode_rlp(calldata_len - 1, rlp_data, items);
+        // eip 1559 tx only for the moment
+        if (tx_type == 2) {
+            // remove the sig to hash the tx
+            let data_len: felt = [items].data_len - SIGNATURE_LEN;
+            let (list_ptr: felt*) = alloc();
+            // add the tx type, see here: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md#specification
+            assert [list_ptr] = tx_type;
+            // encode the rlp list without the sig
+            let (rlp_len: felt) = RLP.encode_rlp_list(data_len, [items].data, list_ptr + 1);
+            let (keccak_ptr: felt*) = alloc();
+            let keccak_ptr_start = keccak_ptr;
+            let (words: felt*) = alloc();
+            // transforms bytes to groups of 64 bits (used for hashing)
+            Helpers.bytes_to_bytes8_little_endian(
+                bytes_len=rlp_len + 1,
+                bytes=list_ptr,
+                index=0,
+                size=rlp_len + 1,
+                bytes8=0,
+                bytes8_shift=0,
+                dest=words,
+                dest_index=0,
+            );
+            // keccak_bigend because verify_eth_signature_uint256 requires bigend
+            let tx_hash = keccak_bigend{keccak_ptr=keccak_ptr}(inputs=words, n_bytes=rlp_len + 1);
+            let (local sub_items: RLP.Item*) = alloc();
+            // decode the rlp elements in the tx (was in the list element)
+            RLP.decode_rlp([items].data_len, [items].data, sub_items);
+            let v = Helpers.bytes_to_felt(sub_items[V_IDX].data_len, sub_items[V_IDX].data, 0);
```

Comment:
can you use here your `is_valid_eth_signature` method? otherwise we have to places that are supposed to do the same thing. Furthermore, since `is_valid_eth_signature`, one expect that this is the only part of the code responsible for this eth signature verification. While it's not used actually for `validate` currently